### PR TITLE
Update toml / geometry format

### PR DESCRIPTION
**Pull request by ahendriksen**
This pull requests

Adds
- A version string to the toml format / `geometry.description' dictionary
- Geometry correction functions in `flexdata.correct'
- A changelog to the toml format / `geometry.description' dictionary
- A `geom.log()' method to write to this changelog
- Many explicit logging functions
- parse_*_log functions to parse and *only parse* log files from the flexray and metadata generated by the script generator. 

Removes:
- read_*_log functions have been deprecated

### CODE REVIEW

File: flexdata/data.py [hunk 1]
```diff
@@ -493,13 +539,28 @@ def read_flexraymeta(path, sample = 1):
...
 
     elif (records['mode'] == 'HW2SW2High'):
         records['det_pixel'] *= 4
         records['img_pixel'] *= 4
+        pixel_adjustment = 4
+
+    if pixel_adjustment != 1:
+       msg = f"Adjusted pixel size by {pixel_adjustement} due to {records['mode']}"
+       logging.info(msg)
+       geom.log(msg)
```
Comment (adriaangraas) [inline flexdata/data.py]: `geom` is undefined here and later in this function.

File: flexdata/correct.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
+import logging
+
+
+###############################################################################
+#                             Correction profiles                             #
+###############################################################################
+profiles = {
+    'cwi-flexray-2019-05-24': {
```
Comment (adriaangraas) [inline flexdata/correct.py]: Should this not be 2019-04-24?

File: flexdata/data.py [hunk 3]
```diff
@@ -435,14 +457,38 @@ def read_flexraylog(path, sample = 1):
...
+    if sample != 1:
+        msg = f"Adjusted geometry by binning by {sample}"
+        logging.info(msg)
+        geom.log(msg)
 
     return geom
 
-def read_flexraymeta(path, sample = 1):
+
+def read_flexraymeta(*args):
```
Comment (adriaangraas) [inline flexdata/data.py]: This function is still called by `read_flexray`.

File: flexdata/data.py [hunk 4]
```diff
@@ -435,14 +457,38 @@ def read_flexraylog(path, sample = 1):
...
+>>> from flexdata import correct
+>>> geom = data.parse_flexraymeta(path, sample=binning)
+>>> geom = correct.correct(geom,
+                           profile='cwi-flexray-2019-05-24',
+                           do_print_changes=True)
+>>> geom = correct.correct_vol_center(geom)
+
+
+""", DeprecationWarning, stacklevel=2)
+   raise NotImplementedError()
```
Comment (adriaangraas) [inline flexdata/data.py]: Indentation is off here.

File: flexdata/data.py [hunk 5]
```diff
@@ -525,6 +592,13 @@ def read_geometry(path, sample = 1):
     records['det_pixel'] *= sample
     records['img_pixel'] *= sample
 
+    # Check version
+    version = records.get('FLEXTOML_VERSION')
+    if version is None:
+       logger.warning(f"No version for toml file. Expected {geometry.FLEXTOML_VERSION}")
+    elif version != geom.FLEXTOML_VERSION:
```
Comment (adriaangraas) [inline flexdata/data.py]: `geom` undefined.

File: flexdata/data.py (hunk 1 above)
Comment (ahendriksen) [reply flexdata/data.py]: Thanks for catching this.

File: flexdata/correct.py (hunk 2 above)
Comment (ahendriksen) [reply flexdata/correct.py]: Yes >.<

File: flexdata/data.py (hunk 4 above)
Comment (ahendriksen) [reply flexdata/data.py]: Will fix this later. It does not crash..

File: flexdata/data.py (hunk 3 above)
Comment (ahendriksen) [reply flexdata/data.py]: Yes, but it only displays a deprecation warning. A function parse_flexraymeta that parses a file has been added.

File: flexdata/data.py (hunk 5 above)
Comment (ahendriksen) [reply flexdata/data.py]: Thanks. Fixed.

**Pull request merged**